server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequestTests.java

```diff
@@ -141,4 +154,88 @@ public void testSerialize() throws Exception {
+        ElasticsearchAssertions.assertToXContentEquivalent(originalBytes, finalBytes, xContentType);
+    }
+
+    public void testUnknownFields() throws IOException {
+        final RolloverRequest request = new RolloverRequest();
+        XContentType xContentType = randomFrom(XContentType.values());
+        final XContentBuilder builder = XContentFactory.contentBuilder(xContentType);
+        builder.startObject();
+        {
+            builder.startObject("conditions");
+            builder.field("max_age", "10d");
+            builder.endObject();
+        }
+        builder.endObject();
+        BytesReference mutated = XContentTestUtils.insertRandomFields(xContentType, builder.bytes(), null, random());
+        expectThrows(ParsingException.class, () -> request.fromXContent(createParser(xContentType.xContent(), mutated)));
+    }
+
+    public void testAddConditions() {
+        RolloverRequest rolloverRequest = new RolloverRequest();
+        Consumer<RolloverRequest> rolloverRequestConsumer = randomFrom(conditionsGenerator);
+        rolloverRequestConsumer.accept(rolloverRequest);
+        expectThrows(IllegalArgumentException.class, () -> rolloverRequestConsumer.accept(rolloverRequest));
+    }
+
+    public void testValidation() {
+        RolloverRequest rolloverRequest = new RolloverRequest();
+        assertNotNull(rolloverRequest.getCreateIndexRequest());
+        ActionRequestValidationException validationException = rolloverRequest.validate();
+        assertNotNull(validationException);
+        assertEquals(1, validationException.validationErrors().size());
+        assertEquals("index alias is missing", validationException.validationErrors().get(0));
+    }
+
+    private static List<Consumer<RolloverRequest>> conditionsGenerator = new ArrayList<>();
+    static {
+        conditionsGenerator.add((request) -> request.addMaxIndexDocsCondition(randomNonNegativeLong()));
+        conditionsGenerator.add((request) -> request.addMaxIndexSizeCondition(new ByteSizeValue(randomNonNegativeLong())));
+        conditionsGenerator.add((request) -> request.addMaxIndexAgeCondition(new TimeValue(randomNonNegativeLong())));
+    }
+
+    private static RolloverRequest createTestItem() throws IOException {
+        RolloverRequest rolloverRequest = new RolloverRequest();
+        if (randomBoolean()) {
+            String type = randomAlphaOfLengthBetween(3, 10);
+            rolloverRequest.getCreateIndexRequest().mapping(type, RandomCreateIndexGenerator.randomMapping(type));
+        }
+        if (randomBoolean()) {
+            RandomCreateIndexGenerator.randomAliases(rolloverRequest.getCreateIndexRequest());
+        }
+        if (randomBoolean()) {
+            rolloverRequest.getCreateIndexRequest().settings(RandomCreateIndexGenerator.randomIndexSettings());
+        }
+        int numConditions = randomIntBetween(0, 3);
+        Set<Consumer<RolloverRequest>> consumers = new HashSet<>();
```

Comment:
While adding the random conditions here seems to work, it took me a while to understand the setup. Maybe this can be simlified a bit by e.g. sampling a random subset from the three options and then using a switch or something? Maybe less elegant but more readable?

Reply:
right, I could have used `randomSubsetOf` and made things much simpler. Will do that.